.github/workflows/ci.yaml

```diff
@@ -393,6 +393,10 @@ jobs:
       - run: pip3 install uvicorn[standard] # For http mirroring test with FastAPI.
       - run: cargo build -p mirrord-layer # Build layer lib. The tests load it into the apps.
       - run: cargo test -p mirrord-layer
+      - uses: actions/setup-node@v3 # version 19 spawns processes with `posix_spawn`, so test that also.
```

Comment:
So is the bug the result of our not supporting node 19? Because I did try to run locally with node 19 and didn't run into any problems.

Reply:
On my machine the bug reproduced when I ran `brew install node` which installed node 19 (alongside a non-homebrew installation I have on my system) and went away whenever I ran `brew uninstall node`. Can it be that even when you installed node 19 yarn used a different version? Or maybe somehow you have a different binary for node 19 than me?
Could you please run again with node 19 and send me the logs?
And what do you get when you run `nm /opt/homebrew/bin/node | grep -e spawn -e exec | grep " U "` (with the path to wherever node 19 is on your system).

Reply:
`node -v` returns v19.5.0, and `which node` returns /opt/homebrew/bin/node
The command you asked me to run returned `                 U _uv_spawn`
Logs are attached:
[logs.log](https://github.com/metalbear-co/mirrord/files/10535412/logs.log)

Reply:
It seems that on your machine, when node spawns `/bin/sh -c node .`, (at least that is what it does not my system), mirrord is loaded into that without hooking and patching SIP. I don't know why this is possible, since `/bin/sh` should probably be SI-Protected. If you'd compile from this branch and run again, we'd have a little more information, since there are a couple of new logs here that say what executables mirrord loads into, with what args etc.

Reply:
See attached.
[logs2.log](https://github.com/metalbear-co/mirrord/files/10537006/logs2.log)

Reply:
Just to make sure, could you please run `ls -lO /bin/sh`? (That's a capital o)

Reply:
We investigated this and it turns out the reason the bug did not reproduce with node 19 for Eyal is that SIP was disabled on his machine.